File: scripts/generate-programs-from-libraries.js
const fs = require('fs');
const path = require('path');

// Charger les données des bibliothèques
const astronomyData = JSON.parse(fs.readFileSync(path.join(__dirname, '../app/data/astronomy-libraries.json'), 'utf8'));
const financeData = JSON.parse(fs.readFileSync(path.join(__dirname, '../app/data/finance-libraries.json'), 'utf8'));

// Fonction pour créer un ID de programme à partir du nom
function createProgramId(name) {
  return name
    .toLowerCase()
    .replace(/[^a-z0-9]/g, '-')
    .replace(/-+/g, '-')
    .replace(/^-|-$/g, '');
}

// Fonction pour créer un programme à partir d'une bibliothèque
function createProgramFromLibrary(library, domain) {
  const programId = createProgramId(library.name);
  
  return {
    id: programId,
    name: library.name.split('/').pop(), // Prendre seulement la dernière partie du nom
    description: `${library.name} - ${domain === 'astronomy' ? 'Astronomy' : 'Finance'} library with ${library.stars} stars`,
    contextFiles: [],
    combinedContextFile: library.hasContextFile ? 
      `/context/${domain}/${library.contextFileName}` : 
      undefined,
    docsUrl: library.github_url,
    extraSystemPrompt: `You are an expert on ${library.name}. Use the provided documentation to help users with this ${domain} library.`
  };
}

// Créer les programmes pour l'astronomie
const astronomyPrograms = astronomyData.libraries.map(lib => 
  createProgramFromLibrary(lib, 'astronomy')
);

// Créer les programmes pour la finance
const financePrograms = financeData.libraries.map(lib => 
  createProgramFromLibrary(lib, 'finance')
);

// Combiner tous les programmes
const allPrograms = [...astronomyPrograms, ...financePrograms];

// Charger la configuration existante
const configPath = path.join(__dirname, '../config.json');
const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));

// Remplacer les programmes existants
config.programs = allPrograms;

// Changer le programme par défaut vers la première bibliothèque d'astronomie
config.defaultProgram = astronomyPrograms[0].id;

// Sauvegarder la nouvelle configuration
fs.writeFileSync(configPath, JSON.stringify(config, null, 2));

console.log(`✅ Généré ${allPrograms.length} programmes:`);
console.log(`   - Astronomy: ${astronomyPrograms.length} programmes`);
console.log(`   - Finance: ${financePrograms.length} programmes`);
console.log(`   - Programme par défaut: ${config.defaultProgram}`);
console.log(`\n📁 Configuration sauvegardée dans: ${configPath}`); 

File: scripts/generate-programs-from-libraries.py
#!/usr/bin/env python3
import json
import re
import os

def create_program_id(name):
    """Créer un ID de programme à partir du nom"""
    return re.sub(r'[^a-z0-9]', '-', name.lower()).replace('-+', '-').strip('-')

def create_program_from_library(library, domain):
    """Créer un programme à partir d'une bibliothèque"""
    program_id = create_program_id(library['name'])
    
    return {
        "id": program_id,
        "name": library['name'].split('/')[-1],  # Prendre seulement la dernière partie du nom
        "description": f"{library['name']} - {domain.title()} library with {library['stars']} stars",
        "contextFiles": [],
        "combinedContextFile": f"/api/context/{domain}/{library['contextFileName']}" if library.get('hasContextFile') else None,
        "docsUrl": library['github_url'],
        "extraSystemPrompt": f"You are an expert on {library['name']}. Use the provided documentation to help users with this {domain} library."
    }

def main():
    # Charger les données des bibliothèques
    with open('app/data/astronomy-libraries.json', 'r') as f:
        astronomy_data = json.load(f)
    
    with open('app/data/finance-libraries.json', 'r') as f:
        finance_data = json.load(f)
    
    # Créer les programmes pour l'astronomie
    astronomy_programs = [create_program_from_library(lib, 'astronomy') 
                         for lib in astronomy_data['libraries']]
    
    # Créer les programmes pour la finance
    finance_programs = [create_program_from_library(lib, 'finance') 
                       for lib in finance_data['libraries']]
    
    # Combiner tous les programmes
    all_programs = astronomy_programs + finance_programs
    
    # Charger la configuration existante
    with open('config.json', 'r') as f:
        config = json.load(f)
    
    # Remplacer les programmes existants
    config['programs'] = all_programs
    
    # Changer le programme par défaut vers la première bibliothèque d'astronomie
    config['defaultProgram'] = astronomy_programs[0]['id']
    
    # Sauvegarder la nouvelle configuration
    with open('config.json', 'w') as f:
        json.dump(config, f, indent=2)
    
    print(f"✅ Généré {len(all_programs)} programmes:")
    print(f"   - Astronomy: {len(astronomy_programs)} programmes")
    print(f"   - Finance: {len(finance_programs)} programmes")
    print(f"   - Programme par défaut: {config['defaultProgram']}")
    print(f"\n📁 Configuration sauvegardée dans: config.json")

if __name__ == "__main__":
    main() 